examples/apps/ai-collab/src/infra/authHelper.ts

```diff
@@ -87,6 +87,44 @@ export async function start(): Promise<{
 	}
 }
 
+export async function getProfilePhoto(): Promise<string> {
+	const msalInstance = await authHelper();
+
+	// Handle the login redirect flows
+	const tokenResponse: AuthenticationResult | null =
+		await msalInstance.handleRedirectPromise();
+
+	// If the tokenResponse is not null, then the user is signed in
+	// and the tokenResponse is the result of the redirect.
+	if (tokenResponse === null) {
+		const currentAccounts = msalInstance.getAllAccounts();
+		if (currentAccounts.length === 0) {
+			// no accounts signed-in, attempt to sign a user in
+			await msalInstance.loginRedirect({
+				scopes: ["FileStorageContainer.Selected", "Files.ReadWrite"],
+			});
+
+			throw new Error(
+				"This should never happen! The previous line should have caused a browser redirect.",
+			);
+		} else {
+			// The user is signed in.
+			// Treat more than one account signed in and a single account the same as this is just a sample.
+			// A real app would need to handle the multiple accounts case.
+			// For now, just use the first account.
+			const account = msalInstance.getAllAccounts()[0];
+			if (account === undefined) {
+				throw new Error("No account found after logging in");
+			}
+			graphHelper = new GraphHelper(msalInstance, account);
+		}
+	} else {
+		graphHelper = new GraphHelper(msalInstance, tokenResponse.account);
```

Comment:
I see `graphHelper` is declared higher in the file. What's the expected use pattern? It's exported, so any module can import and use it. Is there any guarantee the async functions that initialize it run first?

Reply:
Yes, it's declared in higher layer and it's no guarantee other module use it before it gets initialize. However, it will let the fluid client to only login once and then use all permissioned Azure services.